### ISSUE TITLE
Floating filter: onFloatingFilterChanged(null) throws error

### ISSUE THREAD

**Issue by will-tr** [ag-grid/ag-grid#1549]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/ceolter/ag-grid/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Error occurs when setModel(casted.model) is called since var casted = null (no model/apply key)

**Expected behavior**
Filter should be cleared gracefully

**Minimal reproduction of the problem with instructions**
Go to this plnkr which is a fork of Documentation example
Fork: https://plnkr.co/edit/V2VheWqrezGr9CVZ7ag0?p=preview
Example: https://www.ag-grid.com/javascript-grid-date-component/?framework=angularjs#gsc.tab=0

1. In the date column, select any date.
2. In the date column, delete the date text to clear filter
Error should pop up in console

I'm not sure if intent was to write my own onInputBoxChanged to handle the case of when input value is cleared (did this when I wrote my own custom floating filter based on TextFilter), but would be great if baseFilter.onFloatingFilterChanged could handle change === null or empty string

**What is the motivation / use case for changing the behavior?**
Custom date filter in floating filter

**Please tell us about your environment:**

* **ag-Grid version:** 8.2

* **angular version:** 1.6.x

* **Browser:**

"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36"

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
ES5

**Comment by seanlandsman** [comment]
Hi

Thanks for reporting this - I've raised issue AG-359 our side and we'll update this thread with updates as soon as we have them

thanks

**Comment by makinggoodsoftware** [comment]
Hi I am happy to announce that this feature it´s now available in our latest release 9.1.0
